101-hdinsight-hbase-replication-one-vnet/azuredeploy.json

```diff
@@ -0,0 +1,300 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "clusterNamePrefix": {
+      "type": "string",
+      "metadata": {
+        "description": "The prefix of the HDInsight cluster name."
+      }
+    },
+    "clusterLoginUserName": {
+      "type": "string",
+      "defaultValue": "admin",
+      "metadata": {
+        "description": "These credentials can be used to submit jobs to the cluster and to log into cluster dashboards."
+      }
+    },
+    "clusterLoginPassword": {
+      "type": "securestring",
+      "metadata": {
+        "description": "The password must be at least 10 characters in length and must contain at least one digit, one non-alphanumeric character, and one upper or lower case letter."
+      }
+    },
+    "sshUserName": {
+      "type": "string",
+      "defaultValue": "sshuser",
+      "metadata": {
+        "description": "These credentials can be used to remotely access the cluster."
+      }
+    },
+    "sshPassword": {
+      "type": "securestring",
+      "metadata": {
+        "description": "The password must be at least 10 characters in length and must contain at least one digit, one non-alphanumeric character, and one upper or lower case letter."
+      }
+    }
+  },
+  "variables": {
+    "clusterNode1":{
+      "name": "[concat(substring(parameters('clusterNamePrefix'),0,5),'1')]",
+      "defaultStorageAccount":{
+        "name": "[concat(substring(parameters('clusterNamePrefix'),0,5),'1store')]",
+        "type": "Standard_LRS"
+      }
+    },
+    "clusterNode2":{
+      "name": "[concat(substring(parameters('clusterNamePrefix'),0,5),'2')]",
+      "defaultStorageAccount":{
+        "name": "[concat(substring(parameters('clusterNamePrefix'),0,5),'2store')]",
```

Comment:
can we work uniqueString() into the storage account names - limiting the prefix to 5 chars is going to be problematic on a globally unique name.